Melosys 4581 familie steg trygdeavtale

janitamarielle opened the pull request:
Legger til familie-steget i Trygdeavtale:
- Familiesteget er ganske lik FTRLs Familiesteg. Unntaket er i grove trekk: 
  - enableReinitialize = true
  - oppdatereFlyt ved endring (selv når ikke valid)
  - støtte kun for én ektefelle
- boolTilBOOLSK_STRING og BOOLSK_STRINGTilBool
- oppdaterStegData -> oppdaterFlyt
- Ny FamilieMedlem istedenfor å bruke AvklarteFaktas MedfolgendeFamilie, dette siden det er mulig å ha omfatte = undefined her.
- Viser ektefelleSamboerMedPaReisen. Dobbeltsjekker dette med fag

Jeg tester med docker-compose mot mel-api: https://github.com/navikt/melosys-api/pull/807 og mel-trygdeavtale: https://github.com/navikt/melosys-trygdeavtale/pull/12 (profile=local)

janitamarielle committed:
Legger til familie-steget

janitamarielle committed:
Håndterer endringer i menypanel uten å knekke

janitamarielle committed:
Fikser logikk bak oppdatering av steg

janitamarielle committed:
renamer oppdaterStegData -> oppdaterFlyt

janitamarielle committed:
linting og diverse småfiks

janitamarielle committed:
Sender ved all endring, siden det ikke skal være noe logikk i frontend

JohanSakshaug commented on src/felleskomponenter/menypanel/menypanel.tsx:
```diff
@@ -62,7 +62,7 @@ export const Menypanel = ({
       behandlingstema !== MKV.Koder.behandlinger.behandlingstema.BESLUTNING_LOVVALG_NORGE
     ),
-    visEktefelleSamboerMedPaReisen: behandlingstema === ARBEID_I_UTLANDET,
+    visEktefelleSamboerMedPaReisen: behandlingstema === ARBEID_I_UTLANDET || behandlingstema === TRYGDEAVTALE_UK,
```
Test på linje 255 i https://github.com/navikt/melosys-web/blob/33c808169e6fc7051432c68ec6c32baa35774b7c/src/felleskomponenter/menypanel/menypanel.test.tsx#L255
burde oppdateres.

JohanSakshaug commented on src/utils/streng.js:
```diff
@@ -15,10 +16,24 @@ export function boolTilStreng(value) {
+  }
+  return value ? BOOLSK_STRING.SANN : BOOLSK_STRING.USANN;
+}
+
```
Synes det blir ryddigere hvis:

Konsumentene av `boolTilBOOLSK_STRING` kaller `boolTilStreng()` og gjør en `.toUpperCase()` selv, `boolTilBOOLSK_STRING` fjernes.

eller

at `boolTilBOOLSK_STRING` blir til:
```ts
export function boolTilUppercaseStreng(value) {
 if (value === undefined || value === null) {
    return undefined;
  }
 return boolTilStreng(value).toUpperCase();
}
```

janitamarielle replied:
Da lener jeg mot den nederste isåfall. Liker at konsumentene ikke _vet_ hva BOOLSK_STRING inneholder. Dersom denne i fremtiden skulle endre seg er det mindre steder å endre på.

JohanSakshaug commented on src/utils/streng.js:
```diff
@@ -15,10 +16,24 @@ export function boolTilStreng(value) {
+  }
+  return value === BOOLSK_STRING.SANN;
+}
+
```
Samme som kommentar over, kan gjenbruke `strengTilBool` og burde unngå UPPERCASE i funksjonsnavn. `BOOLSK_STRINGTilBool` -> `uppercaseStrengTilBool`?

JohanSakshaug reviewed (commented):
Noen få kommentarer. Skal få kjørt opp med melosys-trygdeavtale også.

JohanSakshaug reviewed (commented):
1. Når jeg velger "Nei" for at et familiemedlem skal innvilges medlemskap,
2. og velger begrunnelse i dropdown/select som kommer opp,
3. men ikke trykker på fritekstfeltet, er "Fortsett"-knapp disabled.
![image](https://user-images.githubusercontent.com/46343153/136341596-feaa25e7-63a4-4320-9046-b733191e02b3.png)

4. Når jeg så trykker i fritekstfeltet uten å skrive inn noe, blir knappen klikkbar.

Burde ikke knappen være klikkbar i punkt 3.? ☝️

janitamarielle commented:
> Burde ikke knappen være klikkbar i punkt 3.? ☝️

Hmm, jo det er vel sant. Det er jo en valgfri fritekst. Skal fikse 👍

Edit: Dette viste seg å være feil i både melosys-web og melosys-trygdeavtale. Løsning er ute på begge stedene nå.

janitamarielle committed:
Typescript error, FormValues

janitamarielle commented:
Jeg vurderer å fjerne alle trygdeavtale-schema-ene siden de nå kun fungerer som en "dobbelsjekk" for Fortsett-knappen. Logikk og validering skjer jo i melosys-trygdeavtale og hovedgrunnen til dette er for å fjerne logikken fra frontend, også da validering. 
Tanker? @JohanSakshaug @martonSkjevelandNav

JohanSakshaug commented:
> Jeg vurderer å fjerne alle trygdeavtale-schema-ene siden de nå kun fungerer som en "dobbelsjekk" for Fortsett-knappen. Logikk og validering skjer jo i melosys-trygdeavtale og hovedgrunnen til dette er for å fjerne logikken fra frontend, også da validering.

Feilmeldingene i schemaene vises vel til SB, for eksempel ARBEIDSLAND_FELT_KREVES i inngangssteget? Ønsker man å fjerne de?

@janitamarielle

janitamarielle commented:
> Feilmeldingene i schemaene vises vel til SB, for eksempel ARBEIDSLAND_FELT_KREVES i inngangssteget? Ønsker man å fjerne de?

Hadde ikke tenkt på disse. Vi vil jo beholde dem 🤔 Takk for innspill, da tror jeg de ligger her inntil videre. Kan evt endres til å kun sjekke enkelt-felt istedenfor sammenhengen mellom dem. Da får man feilmeldingene du nevner, men slipper dobbelt opp av kompleks validering.

janitamarielle committed:
NavneEndring

janitamarielle committed:
QA, tester og rename/gjenbruk av streng-bool funksjonene

janitamarielle committed:
Flytter debounce til stegvelger.tsx. Dette gjør ting mer oversiktlig og fjerner bug med overskriving av steg fra noen av stegene.
Fjerner logikk fra schema, og bruker ikke formIsValid for å sjekke om flyt skal oppdateres da dette skal skje uansett.
Fjernet periodeEndringHandle da denne er overflødig nå.

janitamarielle committed:
Merge branch 'master' of https://github.com/navikt/melosys-web into MELOSYS-4581_familie-steg-trygdeavtale

martonSkjevelandNav commented on src/sider/trygdeavtale/stegKomponenter/vurderingFamilieSchema.js:
```diff
@@ -0,0 +1,8 @@
+import { object } from "yup";
+
+const vurdering_familie = object().shape({
+  barn: object().nullable(),
```
Burde vel validere på både barn og ektefelle at om de ikke er innvilget så må begrunnelse være fylt ut, for det er vel et krav?

janitamarielle replied:
Satt å tenkte på det. For da får man opp "begrunnelse må fylles inn".
Diskuterte med meg selv om det ble for mye "logikk", men kan se hvordan det blir

janitamarielle committed:
Legger til felt-sjekk på familie-schema

JohanSakshaug commented on src/utils/streng.js:
```diff
@@ -16,22 +15,22 @@ export function boolTilStreng(value) {
     return undefined;
   }
-  return value ? BOOLSK_STRING.SANN : BOOLSK_STRING.USANN;
+  return boolTilStreng(value).toUpperCase();
```
Kan droppe sjekken på `undefined` og `null` her, og i stedet gjøre
`return boolTilStreng(value)?.toUpperCase();`

janitamarielle replied:
Hmm, ja det vil jo returnere undefined der if undefiend/null 👍

JohanSakshaug reviewed (approved):
Ser bra ut 👍

janitamarielle committed:
utnytter boolTilStreng bedre
